ckan/public/base/javascript/modules/autocomplete.js

```diff
@@ -186,15 +186,20 @@ this.ckan.module('autocomplete', function (jQuery) {
      *
      * Returns a text string.
      */
-    formatResult: function (state, container, query) {
-      var term = this._lastTerm || null; // same as query.term
+    formatResult: function (state, container, query, escapeMarkup) {
+      var term = this._lastTerm || (query ? query.term : null) || null; // same as query.term
 
       if (container) {
         // Append the select id to the element for styling.
         container.attr('data-value', state.id);
       }
 
-      return state.text.split(term).join(term && term.bold());
```

Comment:
This original line is a bit confusing to me. I'm not sure what it was intended to do. I understand it as:

It seems to always create a 2 element array with empty strings then joins those with the bolded term.
`state.text.split(term) => ['', '']`
`['', ''].join(term && term.bold()) => <b>apples</b>`

logic:
`state.text` and `term` seem to be identical values. So the `state.text` is split on itself (e.g. `'apples'.split('apples')`) which gives an array with 2 empty elements. Then the array is joined by the term and if the term exists, bold it (e.g. `['', ''].join('apples' && 'apples'.bold())`, if the term is null it returns null.

Reply:
"state.text" means full text and "term" is the query text you entered.

eg) state.text : "<script>alert('test');</script>", term : alert
    after split : ["<script>", "('test');</script>"]
    after join : "<script><b>alert</b>('test');</script>"


eg) state.text : "<script>alert('test');</script>", term : null
    after split : ["<script>alert('test');</script>"]
    after join : "<script>alert('test');</script>"